### PR TITLE
[Doc] Add docstrings for trainer module user-facing classes

### PR DESCRIPTION
**Pull request by thvasilo**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/trainer/gsgnn_trainer.py [hunk 1]
```diff
@@ -31,6 +31,12 @@
 class GSgnnTrainer():
     """ Generic GSgnn trainer.
 
+    This class is used as a mixin for classes that implement trainers
+    for various learning tasks at the node and edge level.
+
+    It contains functions that can be used in the implementing classes'
+    `fit` and `eval` functions.
```
Comment (zhjwy9343) [inline python/graphstorm/trainer/gsgnn_trainer.py:38]: Could you add a the end, "If users would like to implement their own trainers, please extend this class and implement both `fit` and `eval` functions."?

File: python/graphstorm/trainer/ep_trainer.py [hunk 2]
```diff
@@ -33,6 +33,15 @@
...
 
+    This class is used to train models for edge prediction tasks,
+    such as edge classification and edge regression.
+
+    It makes use of the functions provided by `GSgnnTrainer`
+    to define two main functions: `fit` that performs the training
+    for the model that is provided when the object is created,
+    and `eval` that evaluates a provided model against test and
+    validation data.
+
```
Comment (zhjwy9343) [inline python/graphstorm/trainer/ep_trainer.py:44]: Could you add an example of how to construct an Trainer instance, and call its fit() method.
You can refer to GS' own use of trainer as an example and simply it into Pseudo code.
You can refer to this PR for the format: https://github.com/awslabs/graphstorm/pull/456/files

File: python/graphstorm/trainer/glem_np_trainer.py [hunk 3]
```diff
@@ -31,12 +31,25 @@
...
+    such as node classification and node regression, using the
+    GLEM framework [1].
+
+    GLEM will iteratively train a GNN and a Language Model in
+    turn, allowing it to make better use of textual information
+    in the graph.
+
+    The input `model` needs to be an instance of `model.node_glem.GLEM`
+
+    [1] https://arxiv.org/abs/2210.14709
```
Comment (zhjwy9343) [inline python/graphstorm/trainer/glem_np_trainer.py:47]: ditto for example

File: python/graphstorm/trainer/np_trainer.py [hunk 4]
```diff
@@ -32,6 +32,15 @@
...
 
+    This class is used to train models for node prediction tasks,
+    such as node classification and node regression.
+
+    It makes use of the functions provided by `GSgnnTrainer`
+    to define two main functions: `fit` that performs the training
+    for the model that is provided when the object is created,
+    and `eval` that evaluates a provided model against test and
+    validation data.
+
```
Comment (zhjwy9343) [inline python/graphstorm/trainer/np_trainer.py:43]: ditto for example

**Comment by zhjwy9343** [comment]
Can you also add an example for the lp_trainer.py?

**Comment by thvasilo** [comment]
> Can you also add an example for the lp_trainer.py?

Done in latest commit.

File: python/graphstorm/trainer/lp_trainer.py [hunk 5]
```diff
@@ -34,16 +34,44 @@
...
+    It makes use of the functions provided by `GSgnnTrainer`
+    to define two main functions: `fit` that performs the training
+    for the model that is provided when the object is created,
+    and `eval` that evaluates a provided model against test and
+    validation data.
 
     Parameters
     ----------
-    model : GSgnnLinkPredictionModelBase
+    model : GSgnnLinkPredictionModel
```
Comment (zhjwy9343) [inline python/graphstorm/trainer/lp_trainer.py:50]: Could you double check the other trainers' input model, and change from **ModelBase to **Model?

**Review by zhjwy9343 — APPROVED** [review]

File: python/graphstorm/trainer/lp_trainer.py (hunk 5 above)
Comment (thvasilo) [reply python/graphstorm/trainer/lp_trainer.py:50]: Yup, fixed in ep_trainer with last commit.

**Pull request merged**